spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -82,7 +91,52 @@ public Object resolve(MethodParameter methodParameter, List<String> words) {
 
 	@Override
 	public ParameterDescription describe(MethodParameter parameter) {
-		throw new UnsupportedOperationException();
+		Parameter jlrParameter = parameter.getMethod().getParameters()[parameter.getParameterIndex()];
+		CliOption option = jlrParameter.getAnnotation(CliOption.class);
+		ParameterDescription result = ParameterDescription.outOf(parameter);
+		result.help(option.help());
+		Optional<String> defaultValue = defaultValueFor(option);
+		if (defaultValue.isPresent()) {
+			result.defaultValue(defaultValue.get());
+		}
+		List<String> keys = Arrays.asList(option.key());
+		result.keys(keys.stream()
+				.filter(key -> !key.isEmpty())
+				.map(key -> CLI_PREFIX + key)
+				.collect(Collectors.toList()));
+		boolean containsEmptyKey = keys.contains("");
+		result.mandatoryKey(!containsEmptyKey);
+		return result;
+	}
+
+	private Optional<String> defaultValueFor(CliOption option) {
+		// CliOption annotations have two default values, one for when the key is specified without a value,
+		// and one when the key isn't specified (e.g. "command --key" vs "command")
+
+		final boolean unspecifiedDefaultDeclared = !CLI_OPTION_NULL.equals(option.unspecifiedDefaultValue());
+		final boolean specifiedDefaultDeclared = !CLI_OPTION_NULL.equals(option.specifiedDefaultValue());
+
+		if (!unspecifiedDefaultDeclared && !specifiedDefaultDeclared) {
```

Comment:
Yep, that's part of the changes I'm adding based on the previous chat we were having. Will be pushing for review soon :)